lib/activerecord/delay_touching/state.rb

```diff
@@ -43,8 +43,13 @@ def more_records?
       end
 
       def add_record(record, *columns)
-        columns << nil if columns.empty? #if no arguments are passed, we will use nil to infer default column
+        # Inferring nil for touch calls with no column specified is creating duplicate DB calls for nested records
+        # Grab the default timestamp columns now as opposed to at write time
+        columns = record.send(:timestamp_attributes_for_update_in_model) if columns.blank?
```

Comment:
Any reason to change `.empty?` to `.blank?` here? If it's just incidental, it might be worth keeping the way it was.